### ISSUE TITLE
Passing eval_set exposes knowledge about its objects?

### ISSUE THREAD

**Issue by fess10** [catboost/catboost#427]
Problem: When I pass an eval_set in `fit` method of CatBoostRegressor and subsequently get predictions on the same dataset, the results differ very much from the case when I do not pass any eval_set during the training phase. Is it expected behaviour? 

NB: In this set, there are lots of objects with feature values unseen in the training set. Does CatBoostRegressor somehow collect information about them during training phase?

catboost version: 0.9.1.1

**Comment by annaveronika** [comment]
During training we by default calculate frequencies of categorical feature values using train and eval sets. This doesn't lead to overfitting, because we don't use label values of the objects, but in many cases it leads to a boost in quality.
If you don't want the eval dataset to be used for calculating frequencies of the features, you can disable it by using counter_calc_method='SkipTest'

**Comment by fess10** [comment]
Thanks, it explains a lot.
